mongomock/collection.py

```diff
@@ -1197,6 +1198,84 @@ def group(self, key, condition, initial, reduce, finalize=None):
         ret_array = ret_array_copy
         return ret_array
 
+    def _get_group_id_fields(self, keys, parent=''):
+        date_operators = [
+            '$dayOfYear',
+            '$dayOfMonth',
+            '$dayOfWeek',
+            '$year',
+            '$month',
+            '$week',
+            '$hour',
+            '$minute',
+            '$second',
+            '$millisecond']
+
+        fields = []
+
+        if isinstance(keys, dict):
+            for k, v in keys.items():
+                if isinstance(v, str):
+                    if k in date_operators:
+                        fields.append((parent + '__' + k + '_' + v).replace('$', ''))
+                    else:
+                        fields.append(v.replace('$', ''))
+                elif isinstance(v, dict):
+                    fields.extend(self._get_group_id_fields(v, parent + '__' + k if parent else k))
+        elif isinstance(keys, str):
+            fields.append(keys.replace('$', ''))
+
+        return fields
+
+    def _add_group_id_fields(self, out_collection, group_func_keys):
+        date_operators = [
+            'dayOfYear',
+            'dayOfMonth',
+            'dayOfWeek',
+            'year',
+            'month',
+            'week',
+            'hour',
+            'minute',
+            'second',
+            'millisecond']
+
+        clear_group_func_keys = []
+        for key in group_func_keys:
+            out_field = key.split('__')[0]
+
+            for doc in out_collection:
+                if key not in doc.keys():
+                    func_field = key.split('__')[1]
+                    func, in_field = func_field.split('_')
+                    out_value = doc.get(in_field)
+
+                    if func in date_operators:
+                        if func == 'dayOfYear':
+                            pass
+                        elif func == 'dayOfMonth':
+                            out_value = out_value.day
+                        elif func == 'dayOfWeek':
+                            out_value = out_value.isoweekday()
+                        elif func == 'year':
+                            out_value = out_value.year
+                        elif func == 'month':
+                            out_value = out_value.month
+                        elif func == 'week':
+                            pass
+                        elif func == 'hour':
+                            out_value = out_value.hour
+                        elif func == 'minute':
+                            out_value = out_value.minute
+                        elif func == 'millisecond':
+                            pass
```

Comment:
The implementation for second would also be trivial or am I missing something here?